compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -462,7 +460,7 @@ def _done_cb(pub_fut: Future):
 
             log.debug("Thread exit")
 
-        def heartbeat() -> None:
+        def heartbeat(active: bool = True) -> Future[None]:
```

Comment:
I see no calls to heartbeat() that uses the Future return value, but is this to be used later?

Reply:
Oof; scared me for a minute.  It's used for the final heartbeat/status report, after the loop is done and the threads have completed:

```python
heartbeat(active=False).result(timeout=5)
```

<img src="https://github.com/user-attachments/assets/a40aa229-0bf1-4a0c-b63c-029144253878" width="400" title="final heartbeat of _main_loop">

Reply:
Oops, I missed that somehow, even after Ctrl-F searching heartbeat(...